Makefile

```diff
@@ -132,6 +133,21 @@ else
 GOBIN=$(shell go env GOBIN)
 endif
 
+# Get latest git hash
+GIT_COMMIT_HASH=$(shell git rev-parse HEAD)
+
+# Get the local image registry required for clusterctl upgrade tests
+LOCAL_IMAGE_REGISTRY ?= localhost:5000
+
+ifeq (${MAKECMDGOALS},test-e2e-clusterctl-upgrade)
+	IMG_TAG=e2e-${GIT_COMMIT_HASH}
+	IMG_REPO=${LOCAL_IMAGE_REGISTRY}/controller
+endif
+
+ifeq (${MAKECMDGOALS},docker-build-e2e)
+	IMG_TAG=e2e-${GIT_COMMIT_HASH}
+endif
```

Comment:
In fact we don't need all dependent targets to inherit these values. I used to use separate variables, but @tuxtof asked me to use IMG_TAG and IMG_REPO: https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix/pull/331#discussion_r1419102887 and https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix/pull/331#discussion_r1419105460

Reply:
But won't setting them here make all dependent targets use these anyway?

Reply:
But by setting them here won't the dependent targets have these values set anyway?

Reply:
No, it will affect precisely the target and not dependent.

Reply:
I'm not sure I follow, this is being set as a global variable and as such will be used in every target that doesn't have a local override. Please correct me if I'm wrong 😅 

The condition `ifeq (${MAKECMDGOALS},docker-build-e2e)` means this `IMG_TAG` value is only being set when precisely `make docker-build-e2e` is run on the command line. If any other goals are specified then this will not be set. Is that actually what you want? By making these target specific variables you will get the same effect and they will be inherited regardless of whether other goals are specified or not, which provides more consistent usage.

Also, using the `MAKECMDGOALS` variable like this is pretty unusual and makes the Makefile harder for others to grok and imo should only be used when strictly necessary, which I'm not sure it is in this case.

Reply:
@jimmidyson The behaviour that you describe with `docker-build-e2e` target is exactly what I expected.
The semantic of the `MAKECMDGOALS` var is very straightforward and imho pretty understandable even if you never used the variable.
We're going to make refactoring work, which I think should include the Makefile and dev scripts. Let's make improvements in separate PRs.